Installation fails on Ubuntu 18.04

I tried `pip3 install wmd` and `pip install git+https://github.com/src-d/wmd-relax`. Result is the same:

```
~ ❯❯❯ pip3 install wmd
Collecting wmd
  Using cached https://files.pythonhosted.org/packages/fe/a1/84d228ca438f1bd5b707045571da9d72bab54c75a87a0585675f4ef5b866/wmd-1.2.8.tar.gz
Collecting numpy (from wmd)
  Using cached https://files.pythonhosted.org/packages/88/29/f4c845648ed23264e986cdc5fbab5f8eace1be5e62144ef69ccc7189461d/numpy-1.15.0-cp36-cp36m-manylinux1_x86_64.whl
Building wheels for collected packages: wmd
  Running setup.py bdist_wheel for wmd ... error
  Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-re6qq18p/wmd/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpjt_slj6dpip-wheel- --python-tag cp36:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.6
  creating build/lib.linux-x86_64-3.6/wmd
  copying wmd/__init__.py -> build/lib.linux-x86_64-3.6/wmd
  running build_ext
  building 'libwmdrelax' extension
  creating build/temp.linux-x86_64-3.6
  creating build/temp.linux-x86_64-3.6/or-tools
  creating build/temp.linux-x86_64-3.6/or-tools/src
  creating build/temp.linux-x86_64-3.6/or-tools/src/graph
  creating build/temp.linux-x86_64-3.6/or-tools/src/base
  creating build/temp.linux-x86_64-3.6/or-tools/src/util
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.6-EKG1lX/python3.6-3.6.5=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/not_a_robot/.local/lib/python3.6/site-packages/numpy/core/include -Ior-tools/src -I/usr/include/python3.6m -c python.cc -o build/temp.linux-x86_64-3.6/python.o -fopenmp -std=c++11 -march=native -ftree-vectorize -DNDEBUG -Wno-sign-compare -flto
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.6-EKG1lX/python3.6-3.6.5=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/not_a_robot/.local/lib/python3.6/site-packages/numpy/core/include -Ior-tools/src -I/usr/include/python3.6m -c or-tools/src/graph/min_cost_flow.cc -o build/temp.linux-x86_64-3.6/or-tools/src/graph/min_cost_flow.o -fopenmp -std=c++11 -march=native -ftree-vectorize -DNDEBUG -Wno-sign-compare -flto
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.6-EKG1lX/python3.6-3.6.5=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/not_a_robot/.local/lib/python3.6/site-packages/numpy/core/include -Ior-tools/src -I/usr/include/python3.6m -c or-tools/src/graph/max_flow.cc -o build/temp.linux-x86_64-3.6/or-tools/src/graph/max_flow.o -fopenmp -std=c++11 -march=native -ftree-vectorize -DNDEBUG -Wno-sign-compare -flto
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.6-EKG1lX/python3.6-3.6.5=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/not_a_robot/.local/lib/python3.6/site-packages/numpy/core/include -Ior-tools/src -I/usr/include/python3.6m -c or-tools/src/base/stringprintf.cc -o build/temp.linux-x86_64-3.6/or-tools/src/base/stringprintf.o -fopenmp -std=c++11 -march=native -ftree-vectorize -DNDEBUG -Wno-sign-compare -flto
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.6-EKG1lX/python3.6-3.6.5=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/not_a_robot/.local/lib/python3.6/site-packages/numpy/core/include -Ior-tools/src -I/usr/include/python3.6m -c or-tools/src/base/logging.cc -o build/temp.linux-x86_64-3.6/or-tools/src/base/logging.o -fopenmp -std=c++11 -march=native -ftree-vectorize -DNDEBUG -Wno-sign-compare -flto
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.6-EKG1lX/python3.6-3.6.5=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/not_a_robot/.local/lib/python3.6/site-packages/numpy/core/include -Ior-tools/src -I/usr/include/python3.6m -c or-tools/src/base/sysinfo.cc -o build/temp.linux-x86_64-3.6/or-tools/src/base/sysinfo.o -fopenmp -std=c++11 -march=native -ftree-vectorize -DNDEBUG -Wno-sign-compare -flto
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.6-EKG1lX/python3.6-3.6.5=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/not_a_robot/.local/lib/python3.6/site-packages/numpy/core/include -Ior-tools/src -I/usr/include/python3.6m -c or-tools/src/util/stats.cc -o build/temp.linux-x86_64-3.6/or-tools/src/util/stats.o -fopenmp -std=c++11 -march=native -ftree-vectorize -DNDEBUG -Wno-sign-compare -flto
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  x86_64-linux-gnu-g++ -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -specs=/usr/share/dpkg/no-pie-link.specs -Wl,-z,relro -Wl,-Bsymbolic-functions -specs=/usr/share/dpkg/no-pie-link.specs -Wl,-z,relro -g -fdebug-prefix-map=/build/python3.6-EKG1lX/python3.6-3.6.5=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.6/python.o build/temp.linux-x86_64-3.6/or-tools/src/graph/min_cost_flow.o build/temp.linux-x86_64-3.6/or-tools/src/graph/max_flow.o build/temp.linux-x86_64-3.6/or-tools/src/base/stringprintf.o build/temp.linux-x86_64-3.6/or-tools/src/base/logging.o build/temp.linux-x86_64-3.6/or-tools/src/base/sysinfo.o build/temp.linux-x86_64-3.6/or-tools/src/util/stats.o -o build/lib.linux-x86_64-3.6/libwmdrelax.cpython-36m-x86_64-linux-gnu.so -flto
  /usr/bin/x86_64-linux-gnu-ld: /tmp/ccoaQoby.ltrans0.ltrans.o: relocation R_X86_64_32S against symbol `_ZTIZN6_pyobjI16tagPyArrayObjectEC4EP7_objectEUlPS0_E_' can not be used when making a shared object; recompile with -fPIC
  /usr/bin/x86_64-linux-gnu-ld: /tmp/ccoaQoby.ltrans1.ltrans.o: relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
  /usr/bin/x86_64-linux-gnu-ld: /tmp/ccoaQoby.ltrans2.ltrans.o: relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a shared object; recompile with -fPIC
  /usr/bin/x86_64-linux-gnu-ld: /tmp/ccoaQoby.ltrans3.ltrans.o: relocation R_X86_64_32S against `.rodata' can not be used when making a shared object; recompile with -fPIC
  /usr/bin/x86_64-linux-gnu-ld: final link failed: Nonrepresentable section on output
  collect2: error: ld returned 1 exit status
  error: command 'x86_64-linux-gnu-g++' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for wmd
  Running setup.py clean for wmd
Failed to build wmd
Installing collected packages: numpy, wmd
  Running setup.py install for wmd ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-re6qq18p/wmd/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-1gympex9-record/install-record.txt --single-version-externally-managed --compile --user --prefix=:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.6
    creating build/lib.linux-x86_64-3.6/wmd
    copying wmd/__init__.py -> build/lib.linux-x86_64-3.6/wmd
    running build_ext
    building 'libwmdrelax' extension
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/or-tools
    creating build/temp.linux-x86_64-3.6/or-tools/src
    creating build/temp.linux-x86_64-3.6/or-tools/src/graph
    creating build/temp.linux-x86_64-3.6/or-tools/src/base
    creating build/temp.linux-x86_64-3.6/or-tools/src/util
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.6-EKG1lX/python3.6-3.6.5=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/not_a_robot/.local/lib/python3.6/site-packages/numpy/core/include -Ior-tools/src -I/usr/include/python3.6m -c python.cc -o build/temp.linux-x86_64-3.6/python.o -fopenmp -std=c++11 -march=native -ftree-vectorize -DNDEBUG -Wno-sign-compare -flto
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.6-EKG1lX/python3.6-3.6.5=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/not_a_robot/.local/lib/python3.6/site-packages/numpy/core/include -Ior-tools/src -I/usr/include/python3.6m -c or-tools/src/graph/min_cost_flow.cc -o build/temp.linux-x86_64-3.6/or-tools/src/graph/min_cost_flow.o -fopenmp -std=c++11 -march=native -ftree-vectorize -DNDEBUG -Wno-sign-compare -flto
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.6-EKG1lX/python3.6-3.6.5=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/not_a_robot/.local/lib/python3.6/site-packages/numpy/core/include -Ior-tools/src -I/usr/include/python3.6m -c or-tools/src/graph/max_flow.cc -o build/temp.linux-x86_64-3.6/or-tools/src/graph/max_flow.o -fopenmp -std=c++11 -march=native -ftree-vectorize -DNDEBUG -Wno-sign-compare -flto
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.6-EKG1lX/python3.6-3.6.5=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/not_a_robot/.local/lib/python3.6/site-packages/numpy/core/include -Ior-tools/src -I/usr/include/python3.6m -c or-tools/src/base/stringprintf.cc -o build/temp.linux-x86_64-3.6/or-tools/src/base/stringprintf.o -fopenmp -std=c++11 -march=native -ftree-vectorize -DNDEBUG -Wno-sign-compare -flto
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.6-EKG1lX/python3.6-3.6.5=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/not_a_robot/.local/lib/python3.6/site-packages/numpy/core/include -Ior-tools/src -I/usr/include/python3.6m -c or-tools/src/base/logging.cc -o build/temp.linux-x86_64-3.6/or-tools/src/base/logging.o -fopenmp -std=c++11 -march=native -ftree-vectorize -DNDEBUG -Wno-sign-compare -flto
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.6-EKG1lX/python3.6-3.6.5=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/not_a_robot/.local/lib/python3.6/site-packages/numpy/core/include -Ior-tools/src -I/usr/include/python3.6m -c or-tools/src/base/sysinfo.cc -o build/temp.linux-x86_64-3.6/or-tools/src/base/sysinfo.o -fopenmp -std=c++11 -march=native -ftree-vectorize -DNDEBUG -Wno-sign-compare -flto
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.6-EKG1lX/python3.6-3.6.5=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/not_a_robot/.local/lib/python3.6/site-packages/numpy/core/include -Ior-tools/src -I/usr/include/python3.6m -c or-tools/src/util/stats.cc -o build/temp.linux-x86_64-3.6/or-tools/src/util/stats.o -fopenmp -std=c++11 -march=native -ftree-vectorize -DNDEBUG -Wno-sign-compare -flto
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    x86_64-linux-gnu-g++ -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -specs=/usr/share/dpkg/no-pie-link.specs -Wl,-z,relro -Wl,-Bsymbolic-functions -specs=/usr/share/dpkg/no-pie-link.specs -Wl,-z,relro -g -fdebug-prefix-map=/build/python3.6-EKG1lX/python3.6-3.6.5=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.6/python.o build/temp.linux-x86_64-3.6/or-tools/src/graph/min_cost_flow.o build/temp.linux-x86_64-3.6/or-tools/src/graph/max_flow.o build/temp.linux-x86_64-3.6/or-tools/src/base/stringprintf.o build/temp.linux-x86_64-3.6/or-tools/src/base/logging.o build/temp.linux-x86_64-3.6/or-tools/src/base/sysinfo.o build/temp.linux-x86_64-3.6/or-tools/src/util/stats.o -o build/lib.linux-x86_64-3.6/libwmdrelax.cpython-36m-x86_64-linux-gnu.so -flto
    /usr/bin/x86_64-linux-gnu-ld: /tmp/ccULccKD.ltrans0.ltrans.o: relocation R_X86_64_32S against symbol `_ZTIZN6_pyobjI16tagPyArrayObjectEC4EP7_objectEUlPS0_E_' can not be used when making a shared object; recompile with -fPIC
    /usr/bin/x86_64-linux-gnu-ld: /tmp/ccULccKD.ltrans1.ltrans.o: relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
    /usr/bin/x86_64-linux-gnu-ld: /tmp/ccULccKD.ltrans2.ltrans.o: relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a shared object; recompile with -fPIC
    /usr/bin/x86_64-linux-gnu-ld: /tmp/ccULccKD.ltrans3.ltrans.o: relocation R_X86_64_32S against `.rodata' can not be used when making a shared object; recompile with -fPIC
    /usr/bin/x86_64-linux-gnu-ld: final link failed: Nonrepresentable section on output
    collect2: error: ld returned 1 exit status
    error: command 'x86_64-linux-gnu-g++' failed with exit status 1
    
    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-re6qq18p/wmd/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-1gympex9-record/install-record.txt --single-version-externally-managed --compile --user --prefix=" failed with error code 1 in /tmp/pip-build-re6qq18p/wmd/
```

I updated python-dev, and installed a bunch of other libraries: libpq-dev, libxml2-dev, libxslt1-dev, libldap2-dev, libsasl2-dev, libffi-dev. It did no help.

I tried to install wmd into virtualenv and into system environment.

Python 3.6.5

--- a/setup.py
+++ b/setup.py
@@ -13,16 +13,16 @@ PACKAGE = "wmd"
 
 CXX_FLAGS = {
    "Darwin": ["-std=c++11", "-march=native", "-ftree-vectorize", "-DNDEBUG",
-              "-Wno-sign-compare", "-flto"],
+              "-Wno-sign-compare", "-fPIC", "-flto"],
    "Linux": ["-fopenmp", "-std=c++11", "-march=native", "-ftree-vectorize",
-             "-DNDEBUG", "-Wno-sign-compare", "-flto"],
+             "-DNDEBUG", "-Wno-sign-compare", "-fPIC", "-flto"],
    "Windows": ["/openmp", "/std:c++latest", "/arch:AVX2", "/DNDEBUG", "/LTCG",
                "/GL"]
 }
 
 LD_FLAGS = {
-    "Darwin": ["-flto"],
-    "Linux": ["-flto"],
+    "Darwin": ["-fPIC", "-flto"],
+    "Linux": ["-fPIC", "-flto"],
     "Windows": ["/LTCG", "/GL"]
 }
 

